lib/kv_storage_redis.ex

```diff
@@ -1,18 +1,116 @@
 defmodule KvStorageRedis do
   @moduledoc """
-  Documentation for `KvStorageRedis`.
+  Модуль для работы с БД Redis
   """
 
+  use GenServer
+
+  defstruct host: nil, port: nil, db: nil, conn: nil
+
+  ### Interface
+
+  @doc ~S"""
+  Запуск процесса
+
+  ## Примеры
+
+      iex> {:ok, _pid} = KvStorageRedis.start()
+
+  """
+  @spec start(any, any, any) :: :ignore | {:error, any} | {:ok, pid}
+  def start(host \\ "127.0.0.1", port \\ 6379, db \\ 0) do
```

Comment:
Это не обязательно, но по конвенции принято называть функцию, которая запускает `GenServer` в дереве супервизоров `start_link/3`, чтобы интерфейс намекал на это, и никто бы не подумал, что это unmonitored `start`.

С типами надо бы поаккуратнее :)
```suggestion
  @spec start_link(binary(), not_neg_integer(), pos_integer()) :: GenServer.on_start()
  def start_link(host \\ "127.0.0.1", port \\ 6379, db \\ 0) do
```
https://hexdocs.pm/elixir/GenServer.html#t:on_start/0